inc/zoo/swar/associative_iteration.h

```diff
@@ -205,6 +210,34 @@ constexpr auto associativeOperatorIterated_regressive(
     return result;
 }
 
+
+// What I don't understand is why this doesn't work?
+template <typename T>
+constexpr auto multiply(T a , T b) {
+    auto operation = [](auto left, auto right, auto count) {
+      if (count) {
+        return left + right;
+      } else {
+        return left;
+      }
+    };
+
+    auto halver = [](auto count) {
+      return count >> 1;
```

Comment:
This is incorrect, this consumes the least significant bit, thus this is a halver for the "progressive" formulation, and you are using it for the "regressive".
The confusion needs me to apologize, I'm calling "halver" to something that in reality ought to double.  I'm trying to describe "the thing that messes with the counts in a way that determines what to do the next iteration"

Reply:
ahhh yes... of course... should have been more attentive to the other ones, and undestood that. thanks!